src/cparsec3/stream/stream_string.c

```diff
@@ -4,70 +4,26 @@
 
 #include <string.h>
 
-Eq(Token(String)) Trait(Eq(Token(String))) {
-  Eq(char) t = trait(Eq(char));
-  return *(Eq(Token(String))*)&t;
-}
-
-Ord(Token(String)) Trait(Ord(Token(String))) {
-  Ord(char) t = trait(Ord(char));
-  return *(Ord(Token(String))*)&t;
-}
-
-Show(Token(String)) Trait(Show(Token(String))) {
-  Show(char) t = trait(Show(char));
-  return *(Show(Token(String))*)&t;
-}
-
-ListT(Token(String)) Trait(List(Token(String))) {
-  ListT(char) t = trait(List(char));
-  return *(ListT(Token(String))*)&t;
-}
-
-Eq(List(Token(String))) Trait(Eq(List(Token(String)))) {
-  Eq(List(char)) t = trait(Eq(List(char)));
-  return *(Eq(List(Token(String))*))&t;
-}
-
-Ord(List(Token(String))) Trait(Ord(List(Token(String)))) {
-  Ord(List(char)) t = trait(Ord(List(char)));
-  return *(Ord(List(Token(String))*))&t;
-}
-
-Iterable(List(Token(String))) Trait(Iterable(List(Token(String)))) {
-  Iterable(List(char)) t = trait(Iterable(List(char)));
-  return *(Iterable(List(Token(String)))*)&t;
-}
-
-ItrT(List(Token(String))) Trait(Itr(List(Token(String)))) {
-  ItrT(List(char)) t = trait(Itr(List(char)));
-  return *(ItrT(List(Token(String)))*)&t;
-}
-
-ArrayT(Token(String)) Trait(Array(Token(String))) {
-  ArrayT(char) t = trait(Array(char));
-  return *(ArrayT(Token(String))*)&t;
-}
-
-Iterable(Array(Token(String))) Trait(Iterable(Array(Token(String)))) {
-  Iterable(Array(char)) t = trait(Iterable(Array(char)));
-  return *(Iterable(Array(Token(String)))*)&t;
-}
-
-ItrT(Array(Token(String))) Trait(Itr(Array(Token(String)))) {
-  ItrT(Array(char)) t = trait(Itr(Array(char)));
-  return *(ItrT(Array(Token(String)))*)&t;
-}
-
 /**
  * Tests whether the stream was empty or not.
  * \param s    a stream
  * \return     true if s was empty, false otherwise.
  */
-static bool empty(String s) {
+static bool null(String s) {
+  assert(s && "Null pointer");
   return !s[0];
 }
 
+static List(Token(String)) chunkToTokens(Tokens(String) chk) {
+  assert(chk && "Null pointer");
+  return trait(List(Token(String))).from_array(strlen(chk), (char*)chk);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Does not handle strings that are not \0-terminated; if given one it may perform an over-read (it could cause a crash if unprotected) (CWE-126).  ](https://app.codacy.com/manual/mori0091/cparsec3/pullRequest?prid=4851446)